docs/root/intro/arch_overview/http/upgrades.rst

```diff
@@ -182,6 +182,52 @@ must be set to terminate ``CONNECT-UDP`` requests.
    :emphasize-lines: 15-16, 19-22, 29-30
    :caption: :download:`terminate_http3_connect_udp.yaml </_configs/repo/terminate_http3_connect_udp.yaml>`
 
+.. _tunneling-udp-over-http:
+
+Tunneling UDP over HTTP
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+   Raw UDP tunneling is in an alpha status and may not be stable enough for use in production.
+   We recommend to use this feature with caution.
+
+Apart from ``CONNECT-UDP`` termination, as described in the section above, Envoy also has support for tunneling raw UDP over HTTP
+``CONNECT`` or HTTP ``POST`` requests, by utilizing the UDP Proxy listener filter. By default, UDP tunneling is disabled, and can be
+enabled by setting the configuration for :ref:`tunneling_config <envoy_v3_api_field_extensions.filters.udp.udp_proxy.v3.UdpProxyConfig.tunneling_config>`.
+
+.. note::
+   Currently, Envoy only supports UDP tunneling over HTTP/2 streams.
```

Comment:
Where does this limitation come from?

Reply:
Not a technical implementation, only since I focused on ``HTTP/2`` implementation for this round. I intend to add HTTP/3 support and tests along the way.

Reply:
Sounds good. Can we say here that h3 is coming soon, or add a TODO?

Reply:
There's a TODO in line 734 (``// TODO(ohadvano): support upstream HTTP/3.``). I don't think writing in the docs themselves that H3 is coming soon would be the correct action